xline/src/storage/compact/mod.rs

```diff
@@ -59,7 +60,15 @@ impl Compactable for Client<Command> {
             physical: false,
         };
         let request_wrapper = RequestWithToken::new_with_token(request.into(), None);
-        let propose_id = generate_propose_id("auto-compactor");
+        let client_id = loop {
+            if let Ok(client_id) = self.get_client_id().await {
+                break client_id;
+            }
+            // we must get client id here
+            warn!("get client id timeout, retrying...");
+        };
```

Comment:
Please set a upper limit for retries.

Reply:
Done, now limited retrying will be moved into `Client`